Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device.h

```diff
@@ -41,22 +40,89 @@ enum IOSReturnCode : s32
   IPC_EESEXHAUSTED = -1016,  // Max of 2 ES handles exceeded
 };
 
-// A struct for IOS ioctlv calls
-struct SIOCtlVBuffer
+struct IOSRequest
 {
-  explicit SIOCtlVBuffer(u32 address);
-
-  const u32 m_Address;
-  u32 Parameter;
-  u32 NumberInBuffer;
-  u32 NumberPayloadBuffer;
-  u32 BufferVector;
-  struct SBuffer
+  u32 address = 0;
+  IPCCommandType command = IPC_CMD_OPEN;
+  u32 fd = 0;
+  explicit IOSRequest(u32 address);
+  virtual ~IOSRequest() = default;
+  void SetReturnValue(s32 new_return_value) const;
```

Comment:
You're perfectly right; I originally wanted to make the return code be specified when replying to a request, similar to how Open() returns an IOS return code.

This is actually a bit tricky because unlike Open(), the other command handlers can choose to delay the reply or even reply asynchronously (EnqueueReply is also directly used in some devices for async replies). So to be honest, I just went the easy way :P

But I agree that it would make more sense to specify the return value *when replying*. What do you think of changing IPCCommandResult to take a return code, and EnqueueReply as well?

(And yes, HandleCommand is the only place where IOS functions are called via IPC.)